test/test_transforms.py

```diff
@@ -1990,5 +1879,118 @@ def test_adjusts_L_mode():
     assert F.adjust_gamma(x_l, 0.5).mode == 'L'
 
 
+def test_rotate():
+    x = np.zeros((100, 100, 3), dtype=np.uint8)
+    x[40, 40] = [255, 255, 255]
+
+    with pytest.raises(TypeError, match=r"img should be PIL Image"):
+        F.rotate(x, 10)
+
+    img = F.to_pil_image(x)
+
+    result = F.rotate(img, 45)
+    assert result.size == (100, 100)
+    r, c, ch = np.where(result)
+    assert all(x in r for x in [49, 50])
+    assert all(x in c for x in [36])
+    assert all(x in ch for x in [0, 1, 2])
+
+    result = F.rotate(img, 45, expand=True)
+    assert result.size == (142, 142)
+    r, c, ch = np.where(result)
+    assert all(x in r for x in [70, 71])
+    assert all(x in c for x in [57])
+    assert all(x in ch for x in [0, 1, 2])
+
+    result = F.rotate(img, 45, center=(40, 40))
+    assert result.size == (100, 100)
+    r, c, ch = np.where(result)
+    assert all(x in r for x in [40])
+    assert all(x in c for x in [40])
+    assert all(x in ch for x in [0, 1, 2])
+
+    result_a = F.rotate(img, 90)
+    result_b = F.rotate(img, -270)
+
+    assert_equal(np.array(result_a), np.array(result_b))
+
+
+@pytest.mark.parametrize('mode', ["L", "RGB", "F"])
+def test_rotate_fill(mode):
+    img = F.to_pil_image(np.ones((100, 100, 3), dtype=np.uint8) * 255, "RGB")
+
+    num_bands = len(mode)
+    wrong_num_bands = num_bands + 1
```

Comment:
This is totally fine as is but just a side note: it's usually good practice to declare a variable in the region where it's used. so in this case we could have declared this one below right before (or within) the `with pytest.raises` block. It's really a nitpick, no need to change anything!